hawk_eye/data_generation/slice_image.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+""" Contains code to slice up an image into smaller tiles. """
+
+import argparse
+import pathlib
+from typing import List, Tuple
+
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+
+from hawk_eye.data_generation import generate_config as config
+
+
+def slice_image(
+    images: List, tile_size: Tuple[int, int], overlap: int, save_dir: pathlib.Path
+) -> None:
+    """ Take in an image and slice it into smaller images.
+    Args:
+        images: A list of paths to images to be sliced.
+        tile_size: The (width, height) of the tiles.
+        overlap: The overlap between adjacent tiles.
+        save_dir: The path to the directory within which to save slices.
+    Returns:
+        None.
+    """
+    for filename in tqdm(images, desc="Slicing images", total=len(images)):
```

Comment:
`tqdm.tqdm`. I don't know if `total=len(images)` is needed here, but I could be wrong.

Reply:
You do, otherwise the entire bar doesn't show up (as in it doesn't say "x% complete" it just shows you how fast it's iterating)